Formula/p/pter.rb

```diff
@@ -0,0 +1,29 @@
+class Pter < Formula
+  include Language::Python::Virtualenv
+
+  desc "Your console and graphical UI to manage your todo.txt file(s)"
+  homepage "https://vonshednob.cc/pter/"
+  url "https://files.pythonhosted.org/packages/cc/3d/6fc7f754cc7f76b4392a781dabc3029bff9cc43efbe59e7a1cd534f51fbb/pter-3.8.0.tar.gz"
+  sha256 "29672889f0a2fe77a327621a436f0a7a08735fea923966d905f9d1e615815334"
+  license "MIT"
+
+  depends_on "python@3.11"
+
+  resource "cursedspace" do
+    url "https://files.pythonhosted.org/packages/cd/3b/72657c9e867dd5034814dcea21b1128a70a1b8427e48c7de8b3b9ea3dd93/cursedspace-1.5.2.tar.gz"
+    sha256 "21043f80498db9a79d5ee1bb52229fd28ad8871a360601c8f9120ff9dadc2aec"
+  end
+
+  resource "pytodotxt" do
+    url "https://files.pythonhosted.org/packages/51/18/a8f4d15eb31bcde441b0ec090c5d97c435beabc9620199e7f90d2f5ad1af/pytodotxt-1.5.0.tar.gz"
+    sha256 "99be359438c52e0c4fc007e11a89f5a03af00fc6851a6ba7070dfe0e00189009"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    assert_equal "pter #{version}", shell_output("#{bin}/pter --version").strip
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Done, hope it is good ! I think you could mention `expect` and `PTY` for the GUI testing part in this advice. Otherwise we're not really testing the main tool!

Sorry that I missed that part in the cookbook. I was looking at how other were tested and saw that most ncurse based library use version or help hooks ^^